packages/api/src/routers/api/sessions.ts

```diff
@@ -1,112 +1,150 @@
 import type { Row } from '@clickhouse/client';
+import { ClickhouseClient } from '@hyperdx/common-utils/dist/clickhouse';
+import { getMetadata } from '@hyperdx/common-utils/dist/metadata';
+import { renderChartConfig } from '@hyperdx/common-utils/dist/renderChartConfig';
 import opentelemetry, { SpanStatusCode } from '@opentelemetry/api';
 import express from 'express';
-import { isNumber, parseInt } from 'lodash';
+import { parseInt } from 'lodash';
 import { serializeError } from 'serialize-error';
+import { z } from 'zod';
+import { validateRequest } from 'zod-express-middleware';
 
-import * as clickhouse from '@/clickhouse';
-import { getTeam } from '@/controllers/team';
+import { getConnectionById } from '@/controllers/connection';
+import { getNonNullUserWithTeam } from '@/middleware/auth';
+import { Source } from '@/models/source';
 import logger from '@/utils/logger';
+import { objectIdSchema } from '@/utils/zod';
 
 const router = express.Router();
 
-router.get('/', async (req, res, next) => {
-  try {
-    const teamId = req.user?.team;
-    const { endTime, q, startTime } = req.query;
-    if (teamId == null) {
-      return res.sendStatus(403);
-    }
-    const startTimeNum = parseInt(startTime as string);
-    const endTimeNum = parseInt(endTime as string);
-    if (!isNumber(startTimeNum) || !isNumber(endTimeNum)) {
-      return res.sendStatus(400);
-    }
+router.get(
+  '/:sessionId/rrweb',
+  validateRequest({
+    params: z.object({
+      sessionId: z.string(),
+    }),
+    query: z.object({
+      sourceId: objectIdSchema,
+      startTime: z.string().regex(/^\d+$/, 'Must be an integer string'),
```

Comment:
super nit: I think zod should support converting to number natively with coerce:
```
z.number({ coerce: true })
```

Reply:
great tips! will update this in the next PR